### PR TITLE
Adopt the CSS changes that introduce the Red Hat font into PatternFly

### PR DESCRIPTION
**Pull request by KKoukiou**
See https://www.patternfly.org/v4/documentation/overview/red-hat-font

### CODE REVIEW

**Comment by garrett** [comment]
Be sure to add a "Fixes #13173" in the commit message. :wink:

**Comment by martinpitt** [comment]
This feels like a trap that's going to happen again, and most definitively also happens with third-party applications. Can we think a little harder about how to apply this centrally?

Can we set the CSS expansion of the class directly in pf-overrides.less?  That would be the most direct approach which should automatically fix cockpit-composer/ostree/podman.

If that's not practical, could we at least set in page.css?

**Comment by garrett** [comment]
@martinpitt: Not really, without duplicating a bunch of PF4 how it is at the current time. Having the class changes a bunch of things, more than just which typeface is used.

See: https://www.patternfly.org/v4/documentation/overview/red-hat-font

**Comment by garrett** [comment]
@martinpitt: We could "fake it" by manually setting the font too, but that wouldn't catch all the rest of the changes.

(Perhaps we should fake it enough so that other parts are at least not using Overpass. That is, if they compile in our overrides or page, depending on where they go... but then they probably won't, so we can't really control what other apps do.)

**Comment by garrett** [comment]
In other words, we cannot handle cases like:

```css
.pf-c-foobar {
  --pf-c-padding: 2rem; 
  padding: var(--pf-c-padding);
}

.pf-m-redhat-font .pf-c-foobar {
  --pf-c-padding: 10rem; 
}
```

...without duplicating all the places where things change in every instance of `.pf-m-redhat-font * {}`... and then we're duplicating a moving target, which will cause other issues in the future.

File: src/base1/patternfly-overrides.less [hunk 1]
```diff
@@ -651,8 +651,4 @@ th > label {
 
 :root {
   font-size: @font-size-base;
```
[the remark this answers is not in the document]
Comment (garrett) [reply src/base1/patternfly-overrides.less]: Considering @martinpitt's comments, perhaps we should re-add the font variables and also add one for the headings too, to make at least the fonts used in 3rd party pages that use our CSS correct.

[the remark this answers is not in the document]
Comment (KKoukiou) [reply src/base1/patternfly-overrides.less]: @garrett @martinpitt this class changes more things than just defining these two variables that I removed. In addition, patternfly-overrides file is not synced automatically, we do it manually, so adding more stuff there just increases the probabilities that things will get out of sync faster.
IMO it's easier to add the class everywhere - even in the external projects than try to add in our overrides everything that this class would do, and carrying duplicated code around the projects manually.

**Comment by KKoukiou** [comment]
> This feels like a trap that's going to happen again, and most definitively also happens with third-party applications. Can we think a little harder about how to apply this centrally?
> 
> Can we set the CSS expansion of the class directly in pf-overrides.less? That would be the most direct approach which should automatically fix cockpit-composer/ostree/podman.
> 
> If that's not practical, could we at least set in page.css?

See comment above.

File: src/base1/patternfly-overrides.less (hunk 1 above)
[the remark this answers is not in the document]
Comment (martinpitt) [reply src/base1/patternfly-overrides.less]: As per IRC, but FTR: Thanks for the explanations! It seems this hack is the cleanest way for now, and it'll go away once PF4 switches the font by itself.

**Comment by garrett** [comment]
Note, mentioned in IRC: PF4 should eventually switch by default to the new font (and everything else that changes).

**Pull request merged**